src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.ShoppingCartDao;
+import com.cinema.exception.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cant add shopping cart: " + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("FROM ShoppingCart "
+                    + "WHERE id = :userId ", ShoppingCart.class);
```

Comment:
join fetch?

Reply:
I don't need it, I have 
```@OneToMany(fetch = FetchType.EAGER)
    private List<Ticket> tickets;``` and it works well

Reply:
👍